app/models/header_lookup.rb

```diff
@@ -61,24 +63,31 @@ def pid_lookup_by_field(term="", field="")
         end
       end
 
-      memoized_terms[term] = term_pid
-      File.write(memoized_file, memoized_terms)
+      @@all_memoized_terms[memoized_vocab][term] = term_pid
+      File.write(memoized_file, @@all_memoized_terms[memoized_vocab])
     end
 
     term_pid == ABSENT_SUBJECT ? nil : term_pid
   end
 
   private
 
-  # read memoized from file, if no file found return blank hash
-  def read_memoized_headers(filepath)
-    begin
-      file = File.read(filepath)
-      # eval will interpret a ruby string passed as an argument
-      # in this case it constructs a hash from the file
-      eval(file)
-    rescue Errno::ENOENT
-      return {}
+  def search_memoized_terms(term, primary_memoized_vocab)
+    secondary_memoized_vocab = @@all_memoized_terms.keys - [primary_memoized_vocab]
+    ordered_memoized_vocabs = [primary_memoized_vocab] + secondary_memoized_vocab
+
+    # if the term can't be found in the vocab assigned to the field, check the other memoized files
+    ordered_memoized_vocabs.each do |memoized_vocab|
+      if term_pid = @@all_memoized_terms[memoized_vocab][term]
+        return term_pid
+      end
     end
+
+    # nothing has been found in ANY memoized file
+    nil
+  end
+
+  def read_memoized_headers(filepath)
+    file = JSON.parse(File.read(filepath))
```

Comment:
minor: can probably get rid of `file` especially since it's not a file at this point but the Json hash.